dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/VmTools.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.dimensiondata.cloudcontrol.domain;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+@AutoValue
+public abstract class VmTools {
+
+   VmTools() {
+   }
+
+   public abstract String type();
+
+   @Nullable
+   public abstract String versionStatus();
+
+   @Nullable
+   public abstract String runningStatus();
+
+   @Nullable
+   public abstract Integer apiVersion();
+
+   public abstract Builder toBuilder();
```

Comment:
For consistency in the domain model it would be worth unifying the model: add the `toBuilder` method to all objects (some are missing) as long as adding the default empty constructor to all them too (some objects are missing it, especially inner ones).